### PR TITLE
Feature/463 update cypress tests

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-463](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-463)

## Main Changes:
* Updated previous tests that were failing due to new changes, most of which were related to duplicate text with the layer names in the Save panel of the Add/Save Data widget.
  * One bug was related to state management on the Monitoring Report page. That issue was resolved, too.
* Added a visual regression test for the Surrounding Features widget.
* Added a test for the `WaterbodyListVirtualized` component on the state Advanced Search page.
* Added a test for the watershed plans table on the _Restore_ tab.
* Added a test for the dischargers toggle table on the _Overview_ tab.
* Added tests for the list of switches in the Save panel of the Add/Save Data widget.

## Steps To Test:
1. Review the tests, run them, and verify they pass.
2. Verify that the Monitoring Report page and the `AccordionItem` components still function as expected, since slight changes were made to both.

### CODE REVIEW

File: app/client/src/components/shared/Accordion.tsx [hunk 1]
```diff
@@ -269,30 +275,32 @@ function AccordionItem({
...
+      }}
+      onKeyUp={(ev) => {
+        if (ev.key === 'Enter') {
+          const newIsOpen = !isOpen;
+          setIsOpen(newIsOpen);
+          onChange(newIsOpen);
+        }
+      }}
+      role="listitem"
+      tabIndex={0}
```
Comment (cschwinderg) [inline app/client/src/components/shared/Accordion.tsx:298]:
```suggestion
      role="listitem"
```
Moving these events up to the div level causes an issue, where clicking on content in the accordion item causes the accordion item to be closed. 

accordion-item-bug

File: app/cypress/e2e/add-data-widget.cy.ts [hunk 2]
```diff
@@ -328,4 +330,70 @@ describe('Add & Save Data Widget', () => {
...
+      }).should('have.attr', 'aria-checked', 'true');
+      cy.findByRole('switch', { name: 'Toggle Boundaries' }).should(
+        'have.attr',
+        'aria-checked',
+        'false',
+      );
+    });
+  });
+
+  it.only("Test that the save panel includes layers added from the widget's other tabs", () => {
```
Comment (cschwinderg) [inline app/cypress/e2e/add-data-widget.cy.ts:361]:
```suggestion
  it("Test that the save panel includes layers added from the widget's other tabs", () => {
```

Need to remove the `only` flag.

File: app/cypress/e2e/state.cy.ts [hunk 3]
```diff
@@ -242,3 +242,25 @@ describe('State page Water Overview tab', () => {
     cy.findByText(text).should('not.exist');
   });
 });
+
+describe('State page Advanced Search tab', () => {
+  it.only('Displays search results in a virtualized list', () => {
```
Comment (cschwinderg) [inline app/cypress/e2e/state.cy.ts:247]:
```suggestion
  it('Displays search results in a virtualized list', () => {
```

Need to remove the `only` flag.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
I'm still running the cypress tests, but I did come across one issue and found a couple of lingering `only` flags.

File: app/client/src/components/shared/Accordion.tsx (hunk 1 above)
Comment (maxdiebold-erg) [reply app/client/src/components/shared/Accordion.tsx:298]: Good catch, I'll revert that. The only reason I moved that up was for the sake of the screen reader, so the label and tab-able item would be in the same place: currently tabbing over the items doesn't read back the location name. I think, then, the `<header>` tag should have the label, and probably a `role="button"`

**Review by cschwinderg — APPROVED** [review]
Everything looks good now! Thanks for making that fix!

**Comment by maxdiebold-erg** [comment]
@cschwinderg I added a couple more tests, not sure if you want to review them before merging this in.

**Pull request merged**